File: tests/fixtures/project_java/src/main/java/utils/Crypto.java
package utils;

import java.security.MessageDigest;
import java.security.SecureRandom;
import java.util.Base64;

public class Crypto {

    public static String hashPassword(String password) {
        if (password == null || password.isEmpty()) {
            throw new IllegalArgumentException("password cannot be empty");
        }
        try {
            byte[] salt = new byte[16];
            new SecureRandom().nextBytes(salt);
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            md.update(salt);
            byte[] hash = md.digest(password.getBytes());
            return Base64.getEncoder().encodeToString(salt) + ":" + Base64.getEncoder().encodeToString(hash);
        } catch (Exception e) {
            throw new RuntimeException("hashing failed", e);
        }
    }

    public static boolean comparePassword(String password, String storedHash) {
        if (password == null || storedHash == null) {
            return false;
        }
        String[] parts = storedHash.split(":");
        if (parts.length != 2) {
            return false;
        }
        try {
            byte[] salt = Base64.getDecoder().decode(parts[0]);
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            md.update(salt);
            byte[] hash = md.digest(password.getBytes());
            return parts[1].equals(Base64.getEncoder().encodeToString(hash));
        } catch (Exception e) {
            return false;
        }
    }
}

File: tests/fixtures/project_rust/src/utils/crypto.rs
use crate::utils::errors::AppError;

pub fn hash_password(password: &str) -> Result<String, AppError> {
    if password.is_empty() {
        return Err(AppError::new("password cannot be empty"));
    }
    // Stub: simple hash
    Ok(format!("hashed_{}", password.len()))
}

pub fn compare_password(password: &str, stored_hash: &str) -> bool {
    if password.is_empty() || stored_hash.is_empty() {
        return false;
    }
    let expected = format!("hashed_{}", password.len());
    expected == stored_hash
}
